File: backend/app/schemas/notification.py
from pydantic import BaseModel
from typing import Optional


class NotificationOut(BaseModel):
    id: str
    userId: str
    type: str
    title: str
    message: str
    transactionId: Optional[str] = None
    isRead: bool
    createdAt: str

    model_config = {"from_attributes": True}


class UnreadCountOut(BaseModel):
    count: int

File: src/types/notification.ts
// ── Notification types ────────────────────────────────────────────────────────

export type NotificationType =
  | 'transaction'
  | 'budget_alert'
  | 'goal'
  | 'system';

export interface Notification {
  id: string;
  userId: string;
  type: NotificationType;
  title: string;
  message: string;
  transactionId?: string;
  isRead: boolean;
  createdAt: string;   // ISO 8601
}

export interface UnreadCount {
  count: number;
}
